source/fundamentals/data-formats/document-data-format-pojo.txt

```diff
@@ -13,17 +13,226 @@ Document Data Format: POJOs
+       private List<String> colors;
+       private Boolean isBlooming;
+       private Float height;
+
+       // public empty constructor needed for retrieving the POJO
+       public Flower() {}
+
+       public Flower(String name, Boolean isBlooming, Float height, List<String> colors) {
+           this.name = name;
+           this.isBlooming = isBlooming;
+           this.height = height;
+           this.colors = colors;
+       }
+
+       public ObjectId getId() {
+           return id;
+       }
+
+       public void setId(ObjectId id) {
+           this.id = id;
+       }
+
+       public String getName() {
+           return name;
+       }
+
+       public void setName(String name) {
+           this.name = name;
+       }
+
+       public Boolean getIsBlooming() {
+           return isBlooming;
+       }
+
+       public void setIsBlooming(Boolean isBlooming) {
+           this.isBlooming = isBlooming;
+       }
+
+       public Float getHeight() {
+           return height;
+       }
+
+       public void setHeight(Float height) {
+           this.height = height;
+       }
+
+       public List<String> getColors() {
+           return colors;
+       }
+
+       public void setColors(List<String> colors) {
+           this.colors = colors;
+       }
+
+       @Override
+       public String toString() {
+           return "Flower [id=" + id + ", name=" + name + ", colors=" + colors + ", isBlooming=" + isBlooming + ", height=" + height + "]";
+       }
+   }
+
+If you are creating your own POJO for storing and retrieving data in MongoDB,
+make sure to follow the following guidelines:
+
+- The POJO class should not implement interfaces or extend classes from a
+  framework.
+- Include all the fields for which you want to store and retrieve data;
+  make sure they are not ``static`` or ``transient``.
+- If you include public getter or setter methods using the
+  `JavaBean naming conventions <https://docs.oracle.com/javase/tutorial/javabeans/writing/properties.html>`__
+  in your POJO, the driver calls them when serializing or deserializing data.
+  If you omit the getter or setter methods for a public property field, the 
+  driver accesses or assigns them directly.
+
+Configure the Driver for POJOs
+------------------------------
+
+To set up the driver to store and retrieve POJOs, we need to specify:
+
+- The ``PojoCodecProvider``, a codec provider that includes
+  :doc:`Codecs </fundamentals/data-formats/codecs>` that define how to
+  encode/decode the data between the POJO and MongoDB document, and which
+  POJO classes or packages that the Codecs should apply to.
+- A :java-docs:`CodecRegistry <apidocs/bson/org/bson/codecs/configuration/CodecRegistry.html>`
+  instance that contains the codec provider and other related information.
+- A ``MongoClient``, ``MongoDatabase``, or ``MongoCollection`` instance
+  configured to use the ``CodecRegistry``.
+- A ``MongoCollection`` instance configured to cast to your POJO document
```

Comment:
Technically, since it's using generics, it's not casting (or at least, the casts are inserted by the compiler).  Better to say: 

> A MongoCollection instance created with the POJO document class bound to the `TDocument` generic type.